### ISSUE TITLE
Fallback recipe doesn't work if --justfile leading to a sub dir provided

### ISSUE THREAD

**Issue by heavelock** [casey/just#1286]
Hey!

I'm quite often using `--unstable` flag with the fallback recipe when working in nested structures.
Let's say that structure of my justfiles is:
```
repo/justfile
repo/project/justfile
```

While `repo/justfile` contains recipe `cleanup`, the `repo/project/justfile` doesn't. Thus, while being in the dir `repo` I wanted to run: 
```
$ just --unstable --justfile project/justfile cleanup
just --unstable --justfile project/justfile cleanup
error: Justfile does not contain recipe `cleanup`.
Did you mean `clean`?
```

However, when I switched working directory to `repo/project` everything works:
```
$ cd project/
$ just --unstable cleanup 
This is empty cleanup

# The same if I needlessly provide a justfile path
$ just --unstable --justfile justfile cleanup 
This is empty cleanup
```

**Comment by runeimp** [comment]
I believe that is working as expected. Is there a reason you need to specify `--justfile project/justfile` specifically in this case? Does `just --unstable project/justfile cleanup` not work?

**Comment by casey** [comment]
Thanks for the report! I tried to reproduce your setup, but wasn't able to reproduce the issue. Here's my shell session:

```
$ tree
.
├── justfile
└── project
    └── justfile

1 directory, 2 files
$ cat justfile
clean:
  echo clean
$ cat project/justfile
cleanup:
  echo cleanup
$ just --unstable --justfile project/justfile cleanup
echo cleanup
cleanup
```

Did I get the setup right? Also, are there any symlinked directories involved?

**Comment by heavelock** [comment]
Hey! Sorry for the delay in providing an answer. 

So to answer @runeimp question:
```
# in root project dir
❯ just --unstable project/justfile cleanup
error: Justfile does not contain recipes `justfile` or `build`.
❯ just --unstable project/ build
error: Justfile does not contain recipe `build`.
```
So just doesn't really like providing path to a justfile without `--justfile`. 

@casey The case you provided is not exactly what I want to achieve. In your example I would need to run:

```
# while in root dir
$ just --unstable --justfile project/justfile clean
# and I expect:
echo clean
clean
```

**Comment by casey** [comment]
> @casey The case you provided is not exactly what I want to achieve. In your example I would need to run:
> 
> ```
> # while in root dir
> $ just --unstable --justfile project/justfile clean
> # and I expect:
> echo clean
> clean
> ```

Ah, okay, thanks for clarifying, now I understand what's going on. The fallback functionality is only turned on if `SearchConfig` is `FromInvocationDirectory`, which is only the case if you don't pass a justfile or path. I don't see why I couldn't also make fallback work with a path, i.e. `just ../ some-recipe`. I'm not sure about making it work when the user passes a justfile. The thought is that if you specify a specific justfile with `--justfile`, then maybe you only want to run a recipe from that justfile, and not fallback to another.

**Comment by heavelock** [comment]
Yeah, I guess that makes sense in most situations. In my case I am specifying a justfile because I have two in a single directory that build two different artifacts out of a shared codebase. Of course it would be optimal just to merge those two together but legacy code has it's own rules.

**Comment by casey** [comment]
I'm on the fence about this, so I'll leave it open for now. Please comment if you feel strongly one way or another!

I'm relaxing the restriction that fallback doesn't work when a directory is provided, i.e. with `just ../foo` or `just foo/bar`: #1348

**Comment by casey** [comment]
Whoops, didn't mean to close.

**Comment by DianaNites** [comment]
I just hit this myself. I sometimes need to call just from within a recipe, but I also want to be robust against being called "distantly" as advised in the documentation https://just.systems/man/en/listing-available-recipes.html so I use `--justfile {{justfile()}}`, or with modules in mind `source_file`

But this issue seems to mean its not possible to both be robust, and call parent recipes, even though they would work fine in bare `just` calls within the directory?

The relaxed restriction for paths doesnt seem to be useful here either, at least i cannot get it to work / combine with actually passing any arguments such as `--list`

I expected this to work because `fallback` is off/unset by default, so if i set it its because I want or even require this justfile to refer to parent justfiles, and I control what is in my justfiles. If backwards compatibility is a concern then the behavior can be a new setting?
